src/seaport.ts

```diff
@@ -130,6 +131,185 @@ export class OpenSeaPort {
+   * corresponding WrappedNFT ERC20 tokens, and then immediately selling those
+   * ERC20 tokens on their corresponding Uniswap exchange.
+   * Emits the `LiquidateAssets` event when the transaction is prompted.
+   * @param param0 __namedParameters Object
+   * @param tokenIds Token IDs of the assets.
+   * @param tokenAddresses Addresses of the tokens' contracts. Must be the same length as `tokenIds`. Each address corresponds with its respective token ID in the `tokenIds` array.
+   * @param accountAddress Address of the user's wallet
+   * @param uniswapSlippageAllowedInBasisPoints The amount of slippage that a user will tolerate in their Uniswap trade; if Uniswap cannot fulfill the order without more slippage, the whole function will revert.
+   */
+  public async liquidateAssets(
+      { tokenIds, tokenAddresses, accountAddress, uniswapSlippageAllowedInBasisPoints }:
+      { tokenIds: string[];
+        tokenAddresses: string[];
+        accountAddress: string;
+        uniswapSlippageAllowedInBasisPoints: number; }
+    ) {
+
+    if (!tokenIds || !tokenAddresses || tokenIds.length != tokenAddresses.length) {
+      throw new Error("The 'tokenIds' and 'tokenAddresses' arrays must exist and have the same length.")
+    }
+
+    // Check if all tokenAddresses match. If not, then we have a mixedBatch of
+    // NFTs from different NFT core contracts
+    const isMixedBatchOfAssets: boolean = !tokenAddresses.every( (val, i, arr) => val === arr[0] )
+
+    const assets: Asset[] = _.zipWith(tokenIds, tokenAddresses, (tokenId, tokenAddress) => {
+      return { tokenAddress, tokenId }
+    })
+
+    this._dispatch(EventType.LiquidateAssets, { assets, accountAddress })
+
+    const gasPrice = await this._computeGasPrice()
+    const txHash = await sendRawTransaction(this.web3, {
+      from: accountAddress,
+      to: WRAPPED_NFT_LIQUIDATION_PROXY_ADDRESS_MAINNET,
+      value: 0,
+      data: WyvernSchemas.encodeCall(getMethod(WrappedNFTLiquidationProxy, 'liquidateNFTs'),
+        [tokenIds, tokenAddresses, isMixedBatchOfAssets, uniswapSlippageAllowedInBasisPoints]),
+      gasPrice
+    }, error => {
+      this._dispatch(EventType.TransactionDenied, { error, accountAddress })
+    })
+
+    await this._confirmTransaction(txHash, EventType.LiquidateAssets, "Liquidating Assets")
+  }
+
+  /**
+   * Purchases a bundle of WrappedNFT tokens from Uniswap and then unwraps them into ERC721 tokens.
+   * Emits the `PurchaseAssets` event when the transaction is prompted.
+   * @param param0 __namedParameters Object
+   * @param numTokensToBuy The number of WrappedNFT tokens to purchase and unwrap
+   * @param amountInEth The estimated cost in wei for tokens (probably some ratio above the minimum amount to avoid the transaction failing due to frontrunning, minimum amount is found by calling UniswapExchange(uniswapAddress).getEthToTokenOutputPrice(numTokensToBuy.mul(10**18));
```

Comment:
this comment looks like it's for wei, when the variable name and logic below is for ETH